src/components/organisms/ChatSidebar/ChatSidebar.tsx

```diff
@@ -40,8 +40,23 @@ export const ChatSidebar: React.FC = () => {
       chatSettings.openedChatType === ChatTypes.PRIVATE_CHAT,
   });
 
+  const widthStorageKey = "chatSideBarWidth";
+  const initWidth = localStorage.getItem(widthStorageKey);
+  let [width, setWidth] = useState<number>(initWidth ? +initWidth : 400);
+
+  function onChangeWidth(width: number) {
+    if (width < 300) width = 300;
+    setWidth(width);
+    localStorage.setItem(widthStorageKey, width.toString());
+  }
+
+  const styles = {
+    width: width + "px",
+    //transition: 'width 0.2s'
+  };
+
   return (
-    <div className={containerStyles}>
+    <div className={containerStyles} style={styles}>
```

Comment:
See the suggestion in the other comment thread in this review about using `re-resizable` first,  as I'm not sure if the above would even be required anymore if we end up using it.

Reply:
I am not a react developer so I am not sure if I am following this..

I do know that react will call render function each time I change the style object, but then React realizes that I am only changing the style, so when it actually updates the DOM, it only touches the div style. So either way, it would not make any perceptible difference, and from a developer's point of view the code is much simpler. 

You can try creating a ref on the parent div and then maybe changing the style through that ref? The render function won't fire in that case, but you should see no difference to the behavior of the UI.

Reply:
> I do know that react will call render function each time I change the style object, but then React realizes that I am only changing the style, so when it actually updates the DOM, it only touches the div style. So either way, it would not make any perceptible difference, and from a developer's point of view the code is much simpler.

@soichih While the end output code result doesn't change, it requires React (and thus the user's browser/computer) to do more processing work. This is a basic React performance pattern/best practice.

---

> You can try creating a ref on the parent div and then maybe changing the style through that ref? The render function won't fire in that case, but you should see no difference to the behavior of the UI.

@soichih I'm not sure I follow you here? Why would we do that? As an alternative solution to the issue I called out here? Or to demonstrate something about your point?

Reply:
Yes, React will need to do more work but it is so insignificant and user will not notice any difference. I was just suggesting to use `ref=` based style setting so you can actually benchmark the impact of letting React do "more" work

Reply:
@0xdevalias please update if it is still pertinent in current diff or could be marked resolved

Reply:
> Yes, React will need to do more work but it is so insignificant and user will not notice any difference. I was just suggesting to use ref= based style setting so you can actually benchmark the impact of letting React do "more" work

@soichih In the interest of landing these feature/etc PRs, we should follow our established patterns in the codebase, based on best practices. If you want to open a separate issue/PR that benchmarks things and makes a case for why those best practices shouldn't be followed/make now sense/etc, by all means please do, and I will look at it when I have a chance. But until that time, we have our patterns and best practices for a reason, and contributions (whether internal or otherwise) need to stick to those.

Reply:
> please update if it is still pertinent in current diff or could be marked resolved

@yarikoptic I haven't had the opportunity to review the changes in this PR, so I can't say currently one way or the other on that.

Reply:
I've replaced my own implementation of Resizer component with `re-resizable` as @0xdevalias has suggested. This PR no longer touches `style={style}` so I don't see what else must be done for this to get this PR approved. However, as far as I know .. re-resizable is implemented more/less the same way that I was doing - using style={style} with a few mouse events bound to window .. if @0xdevalias doesn't like that.. he should open a github issue on `re-resizable` or reimplement it with according to his coding styles/guideline, etc..

Reply:
I've merged staging, tested -- still seems to work very nicely.  It would be pity to not have this feature for OHBM.

So what would be your verdict @0xdevalias given that it seems that all major comments were addressed and any additional change would require changes to resizable itself if I got it right?

Reply:
Dear @0xdevalias and @mike-lvov , please have a look at this long standing PR.  IMHO all addressable feedback was addressed, and this little feature would be a great boost to UX.